app/gen-server/lib/HomeDBManager.ts

```diff
@@ -741,6 +742,11 @@ export class HomeDBManager extends EventEmitter {
         user.options = {...(user.options ?? {}), authSubject: userOptions.authSubject};
         needUpdate = true;
       }
+      const today = moment().startOf('day');
+      if (today !== moment(user.lastConnectionAt).startOf('day')) {
```

Comment:
Are you sure that will work, if `user.lastConnectionAt` is null ?

Reply:
When I test it in local it works fine but I added a test to be totaly sure of that if you think it's safer

Reply:
But I think (haven't checked that), that moment actually returns objects, so this comparision in this line looks wrong. For example:
```
moment(1).startOf('day') === moment(1).startOf('day') // I think this is false
```
So this condition here, won't ever work, as it will always be false. Maybe there is some magic with `Moment<Invaliddate>`, that reuses the same references, but I think it would be better to make it more obvious, and easier to read.

So I'd refactor this with an explicit check if the lastConnectionAt is null. And reuse `moment.diff` method to make the actual comparison.

Reply:
Yes I see that in fact and I change the code in my last commit, see it here : https://github.com/gristlabs/grist-core/pull/935/files#diff-813f675a1a7eca36983d2e8c0a146953a23292633474d71e78bf0732129e318eR437